### PR TITLE
[HOPSWORKS-1231] Add documentation for SageMaker

### PR DESCRIPTION
**Pull request by steffengr**

### CODE REVIEW

File: docs/user_guide/hopsworks/featurestore.rst [hunk 1]
```diff
@@ -454,6 +454,68 @@ A feature store consists of five main components:
 
     Feature Store Component Hierarchy.
 
+Connecting from Amazon SageMaker
+------------------------------------
```
Comment (tkakantousis) [inline docs/user_guide/hopsworks/featurestore.rst]: @steffengr length needs to match title length

File: docs/user_guide/hopsworks/featurestore.rst [hunk 2]
```diff
@@ -454,6 +454,68 @@ A feature store consists of five main components:
...
+Connecting to the Feature Store from Amazon SageMaker requires a Feature Store API key to be stored in the AWS Secrets Manager. Additionally, read access to this API key needs to be give to the AWS role used by SageMaker and hops-util-py needs to be installed on SageMaker.
```
Comment (tkakantousis) [inline docs/user_guide/hopsworks/featurestore.rst]: `give` -> `given`

File: docs/user_guide/hopsworks/featurestore.rst [hunk 3]
```diff
@@ -454,6 +454,68 @@ A feature store consists of five main components:
...
+
+**Generating an API Key and storing it in the AWS Secrets Manager**
+
+Log in to your Hopsworks installation, click on your username in the top-right corner and select *Settings* to open the user settings. Select *Api keys*. Give the key a name and select the *featurestore* and *project* scopes before creating the key. Copy the key into your clipboard for the next step.
```
Comment (tkakantousis) [inline docs/user_guide/hopsworks/featurestore.rst]: @steffengr `Log in to Hopsworks`. `installation` is redundant.

File: docs/user_guide/hopsworks/featurestore.rst [hunk 4]
```diff
@@ -454,6 +454,68 @@ A feature store consists of five main components:
...
+To be able to access the Hopsworks Feature Store the hops-util-py library needs to be installed. One way of achieving this is opening a Python notebook in SageMaker and installing the latest hops-util-py together with a native depdendency using the following command. This requires root access on the notebook. Note that the library will not be persistet. For information around how to permanently install a library to Sagemaker see `Install External Libraries and Kernels in Notebook Instances <https://docs.aws.amazon.com/sagemaker/latest/dg/nbi-add-external.html>`_. ::
+
+    !sudo yum -y install cyrus-sasl-devel && pip install git+https://github.com/logicalclocks/hops-util-py.git
+
+You can now connect to the Feature Store::
+
+    import hops.featurestore as fs
+    fs.connect('my_instance.us-east-2.compute.amazonaws.com', 'my_project')
+
+If you have troubles connecting then ensure that the Security Group of your Hopsworks instance on AWS are configured to allow incoming traffic from your SageMaker instance. See `VPC Security Groups <https://docs.aws.amazon.com/vpc/latest/userguide/VPC_SecurityGroups.html>`_. for more information.
```
Comment (tkakantousis) [inline docs/user_guide/hopsworks/featurestore.rst]: @steffengr `troubles` -> `trouble` and add a comma after `connecting`

File: docs/user_guide/hopsworks/featurestore.rst [hunk 5]
```diff
@@ -454,6 +454,68 @@ A feature store consists of five main components:
...
+.. _hopsworks_aws_policy.png: ../../_images/aws_policy.png
+.. figure:: ../../imgs/feature_store/aws_policy.png
+    :alt: Hopsworks feature store set policy
+    :target: `hopsworks_aws_policy.png`_
+    :align: center
+    :scale: 20 %
+    :figclass: align-center
+
+**Installing hops-util-py and connecting to the Feature Store**
+To be able to access the Hopsworks Feature Store the hops-util-py library needs to be installed. One way of achieving this is opening a Python notebook in SageMaker and installing the latest hops-util-py together with a native depdendency using the following command. This requires root access on the notebook. Note that the library will not be persistet. For information around how to permanently install a library to Sagemaker see `Install External Libraries and Kernels in Notebook Instances <https://docs.aws.amazon.com/sagemaker/latest/dg/nbi-add-external.html>`_. ::
```
Comment (tkakantousis) [inline docs/user_guide/hopsworks/featurestore.rst]: @steffengr `persistet` -> `persistent`

File: docs/user_guide/hopsworks/featurestore.rst (hunk 4 above)
Comment (steffengr) [reply docs/user_guide/hopsworks/featurestore.rst]: Done I should get a spell checker for my editor.

File: docs/user_guide/hopsworks/featurestore.rst (hunk 5 above)
Comment (steffengr) [reply docs/user_guide/hopsworks/featurestore.rst]: Done

File: docs/user_guide/hopsworks/featurestore.rst (hunk 2 above)
Comment (steffengr) [reply docs/user_guide/hopsworks/featurestore.rst]: Fixed

File: docs/user_guide/hopsworks/featurestore.rst (hunk 1 above)
Comment (steffengr) [reply docs/user_guide/hopsworks/featurestore.rst]: Fixed

**Pull request merged**